src/pathfora.js

```diff
@@ -694,6 +705,40 @@
     },
 
     /**
+     * @description check if a manualTrigger widget
+     * is ready to be displayed, and if so display the widget
+     * @param {object} widget
+     */
+    triggerWidget: function (widget) {
+      var valid;
+
+      for (var key in widget.watchers) {
+        if (widget.watchers.hasOwnProperty(key) && widget.watchers[key] !== null) {
+          valid = widget.valid && widget.watchers[key].check();
+        }
+      }
+
+      if (widget.displayConditions.impressions && valid) {
+        valid = core.impressionsChecker(widget.displayConditions.impressions, widget);
+      }
+
+      if (valid) {
+        context.pathfora.showWidget(widget);
+        widget.valid = false;
+        context.pathfora.triggeredWidgets[widget.id] = false;
+
+        // remove from the ready widgets list
+        core.readyWidgets.forEach(function (w, i) {
+          if (w.id === widget.id) {
+            core.readyWidgets.splice(i, 1);
+          }
+        });
```

Comment:
Splicing the array you are `forEach`ing over like this is prone to bugs, since the indices of elements are going to change.

Fortunately there will only ever be one widget to splice, so you could instead use `Array#some` to exit early.

```js
core.readyWidgets.some(function (w, i) {
  if (w.id === widget.id) {
    core.readyWidgets.splice(i, 1);
    return true;
  }
});
```